modules/weaver-tests/src/smithy4s/weavertests/WeaverHttpResponseTestCase.scala

```diff
@@ -0,0 +1,218 @@
+  * 4. Assertion are performed on the resulting output
+  * 
+  * For the server:
+  * 1. The HttpRequestTestCase#params are used to generated an output, a dummy input is generated via the schema.
+  * 2. The server implementation is setup to a) return the dummy output and b) record the outgoing response for assertion
+  * 3. The dummy input is fed into the a client call (the operation under test). That
+  *    client is hooked to the server setup at step 2. 
+  * 4. Assertion are performed on the recorded response
+  *
+  */
+private class WeaverHttpResponseTestCase[
+    Alg[_[_, _, _, _, _]],
+    Op[_, _, _, _, _]
+](
+    client: (
+        HttpApp[IO],
+        Uri
+    ) => Either[UnsupportedProtocolError, smithy4s.Monadic[Alg, IO]],
+    server: smithy4s.Monadic[Alg, IO] => Either[
+      UnsupportedProtocolError,
+      HttpRoutes[IO]
+    ]
+)(implicit service: Service[Alg, Op], ce: CompatEffect)
+    extends Expectations.Helpers {
+  import ce._
+  import org.http4s.implicits._
+  private val baseUri = uri"http://localhost/"
+
+  type R[I_, E_, O_, SE_, SO_] = IO[O_]
+
+  private def matchResponse(
+      response: Response[IO],
+      testCase: HttpResponseTestCase
+  )(implicit loc: SourceLocation) = {
+    val bodyAssert = testCase.body
+      .map { expectedBody =>
+        response.bodyText.compile.string.map { responseBody =>
+          assert.eql(expectedBody, responseBody)
+        }
+      }
+      .getOrElse(success.pure[IO])
+
+    val headerAssert =
+      testCase.headers
+        .map { expectedHeaders =>
+          expectedHeaders
+            .map { case (name, value) =>
+              val actual = response.headers.get(CIString(name))
+              actual
+                .map { values =>
+                  val actualValue = values.map(_.value).fold
+                  assert.eql(value, actualValue)
+                }
+                .getOrElse(
+                  failure(s"Header $name was not found in the response.")
+                )
+            }
+            .reduce(_ && _)
+        }
+        .getOrElse(success)
+
+    val statusAssert = assert.eql(testCase.code, response.status.code)
+
+    List(
+      bodyAssert,
+      headerAssert.pure[IO],
+      statusAssert.pure[IO]
+    ).sequence.map(_.reduce(_ && _))
+  }
+
+  private[weavertests] def makeServerTest[I, E, O, SE, SO](
+      endpoint: Endpoint[Op, I, E, O, SE, SO],
+      testCase: HttpResponseTestCase,
+      outputFromDocument: Document.Decoder[O]
+  ): GeneratedTest = GeneratedTest(
+    name = endpoint.id.toString + "(server|response): " + testCase.id,
+    assertions = {
+      type R[I_, E_, O_, SE_, SO_] = IO[O_]
+
+      val dummyInput = DefaultSchemaVisitor(endpoint.input)
+      val output =
+        testCase.params
+          .map {
+            outputFromDocument
+              .decode(_)
+              .liftTo[IO]
+          }
+          .getOrElse(IO.unit.asInstanceOf[IO[O]])
```

Comment:
same idea: I'd decode an empty object `Document`

Reply:
empty object is the same a `Unit`? I would see object  being rendered as `{}` where unit is ``

Reply:
> empty object is the same a Unit? 

It is indeed.